jdaviz/app.py

```diff
@@ -1039,6 +1036,11 @@ def add_data_to_viewer(self, viewer_reference, data_path,
             selected_data_items[data_id] = 'visible'
             self._update_selected_data_items(viewer_item.get('id'), selected_data_items)
         else:
+            # This block provides backward compatibility for version<=3.0, where the second arg
+            # in `add_to_viewer` was `data_path` instead of `data_label`. When `data_label` is
+            # a file path and that file exists, load its data
+            if os.path.exists(data_label):
+                self.load_data(data_label)
```

Comment:
*If* we want this, should it also make a call back to `add_data_to_viewer`?  Does `load_data` vs `load_spectrum` matter for the specviz case?  Alternatively we could just suggest calling `load_data` in the error message.

Reply:
So my original suggestion was to throw an exception here to guide the user to call `load_data` themselves. Something along the lines of "Did you mean X?" I'm not sure we want to try and fix the user's call here, in case then they think this is THE way to load data. Going back to Pey-Lian's original argument "we shouldn't have multiple ways to do the same thing"

Reply:
I agree with both of your points. I'll edit this line to raise an error rather than to try loading the data at the specified path.